theano/tensor/type.py

```diff
@@ -204,8 +204,8 @@ def filter_variable(self, other, allow_convert=True):
         Convert a symbolic Variable into a TensorType, if compatible.
 
         For the moment, only a TensorType or CudaNdarrayType will be
```

Comment:
While you're there, GpuArrayType will convert here too.  As will scalars, I think.